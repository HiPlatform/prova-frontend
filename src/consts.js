export const URLS = {
  SEARCH: 'https://api.spotify.com/v1/search',
};

export const CREDENTIALS = {
  headers: {
    Authorization: 'Bearer BQDGGAag6Hocue0iXDFWp0O9CdmjCSq6nI9R2-HE-_N-skmHG8V3btyKOZnilxWjn7cpkFMAbM70G8CH_-Y',
  },
};

export const HOMEPAGE_ACTIONS = {
  SEARCH_DATA: 'SEARCH_DATA',
  SEARCH_TYPE: 'SEARCH_TYPE',
  SEARCH_CHANGE: 'SEARCH_CHANGE',
  SEARCH_DETAILS: 'SEARCH_DETAILS',
  NAVIGATE: 'NAVIGATE',
};
