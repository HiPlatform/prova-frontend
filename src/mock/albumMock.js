export default {
  albums: {
    items: [
      {
        album_type: 'album',
        artists: [
          {
            external_urls: {spotify: 'https://open.spotify.com/artist/2VsqwYdmf0lPQzBeVB9kCk'},
            href: 'https://api.spotify.com/v1/artists/2VsqwYdmf0lPQzBeVB9kCk',
            id: '2VsqwYdmf0lPQzBeVB9kCk',
            name: 'Mock',
            type: 'artist',
            uri: 'spotify:artist:2VsqwYdmf0lPQzBeVB9kCk',
          },
        ],
        available_markets: [
          'AD', 'AR', 'AT', 'AU', 'BE', 'BG', 'BO', 'BR', 'CA', 'CH', 'CL', 'CO', 'CR',
          'CY', 'CZ', 'DE', 'DK', 'DO', 'EC', 'EE', 'ES', 'FI', 'FR', 'GB', 'GR', 'GT',
          'HK', 'HN', 'HU', 'ID', 'IE', 'IL', 'IS', 'IT', 'JP', 'LI', 'LT', 'LU', 'LV',
          'MC', 'MT', 'MX', 'MY', 'NI', 'NL', 'NO', 'NZ', 'PA', 'PE', 'PH', 'PL', 'PT',
          'PY', 'RO', 'SE', 'SG', 'SK', 'SV', 'TH', 'TR', 'TW', 'US', 'UY', 'VN', 'ZA',
        ],
        href: 'https://api.spotify.com/v1/albums/1YwG7cWRl5eJeRlwuFOcG5',
        images: [
          { url: 'https://i.scdn.co/image/5ca67102f3dd4144bc7c41ee474af0ac8c3eed4e' },
          { url: 'https://i.scdn.co/image/3f4e743e38bafba8917b363a4c5f924964f7959f' },
          { url: 'https://i.scdn.co/image/ecc2b56b5ede9ca8d5438526b1f1f71112b4300b' },
        ],
        name: 'Mock',
        type: 'album',
      },
    ],
  },
};
