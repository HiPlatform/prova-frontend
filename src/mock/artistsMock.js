export default {
  artists: {
    items: [
      {
        genres: ['string folk'],
        href: 'https://api.spotify.com/v1/artists/5oW7xntz397IxAiDinC6Tk',
        images: [
          { url: 'https://i.scdn.co/image/4be2260d1e012c613f26d943b1788cc2a4c97e65' },
          { url: 'https://i.scdn.co/image/769dda96acd0d7fbf09b487fa34d283784fb3de8' },
          { url: 'https://i.scdn.co/image/9020f047835ed5ccac3e40ddef4ae15fbfa0cc1d' },
        ],
        name: 'John Mock',
        popularity: 26,
        type: 'artist',
      },
      {
        genres:['emo', 'emo punk'],
        href: 'https://api.spotify.com/v1/artists/5uDuKeTdnkTHdHIGrQWhOL',
        images: [
          { url: 'https://i.scdn.co/image/97bbdad7710788078378bd4d74538d6e88d120a9' },
          { url: 'https://i.scdn.co/image/789efd9cfba28a3b594084bf3be4d18061317201' },
          { url: 'https://i.scdn.co/image/d19e7fc70c44eec288fe518098fe9a853e3d97a0' },
          { url: 'https://i.scdn.co/image/ec87049d3415a82a575feaaf01d4aca6505b10f5' }
        ],
        name: 'Mock Orange',
        popularity: 23,
        type: 'artist',
      },
    ],
  },
};
