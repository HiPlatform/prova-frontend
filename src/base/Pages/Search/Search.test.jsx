import React from 'react';
import { shallow } from 'enzyme';
import { shallowWithStore } from 'enzyme-redux';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { partial } from 'ramda';

import Connected, { Search } from './index';
import getDetails from './Search.actions';
import { SEARCH_ACTIONS, SEARCH_INITIALSTATE, HOME_INITIALSTATE } from '../../../consts';
import SearchReducer from './Search.reducer';
import artistsMock from '../../../mock/artistsMock';
import albumMock from '../../../mock/albumMock';
import trackMock from '../../../mock/trackMock.json';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

describe('Search >', () => {
  const wrapper = shallow(<Search />);
  const home = {
    ...HOME_INITIALSTATE,
    items: artistsMock.artists.items,
  };
  const store = mockStore({ search: SEARCH_INITIALSTATE, home });
  const connectedWrapper = shallowWithStore(<Connected />, store);
  const randomAction = { type: 'random' };
  const detailsAction = getDetails('https://some.redirect/url');

  it('should render without crashing', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('[Connected] Search >', () => {
    it('should render a list with artists cards', () => {
      const cards = connectedWrapper.render().find('.searchPage__content').children();
      expect(cards).toHaveLength(2);
    });

    it('should render a list with album cards', () => {
      const newHome = {
        ...HOME_INITIALSTATE,
        items: albumMock.albums.items,
      };
      const newStore = mockStore({ search: SEARCH_INITIALSTATE, home: newHome });
      const albumWrapper = shallowWithStore(<Connected />, newStore);

      const cards = albumWrapper.render().find('.searchPage__content').children();
      expect(cards).toHaveLength(1);
    });

    it('should render a list with track cards', () => {
      const newHome = {
        ...HOME_INITIALSTATE,
        items: trackMock.tracks.items,
      };
      const newStore = mockStore({ search: SEARCH_INITIALSTATE, home: newHome });
      const trackWrapper = shallowWithStore(<Connected />, newStore);

      const cards = trackWrapper.render().find('.searchPage__content').children();
      expect(cards).toHaveLength(20);
    });

    it('should render not render a list', () => {
      const newHome = {
        ...HOME_INITIALSTATE,
        items: [
          { type: 'mock' },
        ],
      };
      const newStore = mockStore({ search: SEARCH_INITIALSTATE, home: newHome });
      const mockWrapper = shallowWithStore(<Connected />, newStore);

      const cards = mockWrapper.render().find('.searchPage__content').children();
      expect(cards).toHaveLength(0);
    });
  });

  describe('[Actions] SearchPage >', () => {
    describe('getDetails >', () => {
      beforeEach(() => {
        store.clearActions();
        fetch.resetMocks();
        fetch.mockResponse(JSON.stringify({ foo: 'bar' }));
      });

      it('should call the api and get the details', () => {
        const expectedResult = [{ type: SEARCH_ACTIONS.SEARCH_DETAILS, details: { foo: 'bar' } }];

        return store.dispatch(detailsAction).then(() => {
          expect(store.getActions()).toEqual(expectedResult);
        });
      });
    });
  });

  describe('[Reducer] SearchPage >', () => {
    const reducer = partial(SearchReducer, [SEARCH_INITIALSTATE]);

    it('has default called', () => {
      expect(reducer(randomAction)).toEqual(SEARCH_INITIALSTATE);
    });

    it('gets the details of the clicked card', () => {
      const details = { type: SEARCH_ACTIONS.SEARCH_DETAILS, details: { foo: 'bar' } };
      const expectedResult = { ...SEARCH_INITIALSTATE, details: { foo: 'bar' } };

      expect(reducer(details)).toEqual(expectedResult);
    });
  });
});
