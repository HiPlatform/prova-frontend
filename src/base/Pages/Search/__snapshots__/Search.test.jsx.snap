// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search > should render without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    fnDetails={[Function]}
    fnNavigate={[Function]}
    items={Array []}
    searchType=""
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Topbar
          navigate={[Function]}
/>,
        <div
          className="searchPage__content"
>
          
</div>,
      ],
      "className": "searchPage",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "navigate": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [],
          "className": "searchPage__content",
        },
        "ref": null,
        "rendered": Array [],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Topbar
            navigate={[Function]}
/>,
          <div
            className="searchPage__content"
>
            
</div>,
        ],
        "className": "searchPage",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "navigate": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [],
            "className": "searchPage__content",
          },
          "ref": null,
          "rendered": Array [],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
